include/swift/SIL/SILInstruction.h

```diff
@@ -7771,10 +7771,16 @@ class GraphOperationInst final
   unsigned NumOperands;
   /// The attributes of the graph operation.
   MutableArrayRef<GraphOperationAttribute> Attributes;
+  /// When true, this instruction is to be executed out-of-graph (via eager op
+  /// dispatch). Otherwise, we attempt to "cluster" this graph op with other
+  /// graph ops into a graph function. For an op with dynamic attribute(s), it
+  /// must run out-of-graph, and thus have this field set to true.
+  // FIXME: Extend parser and printer with this field.
+  bool NoClustering;
```

Comment:
Don't mean to bikeshed, but would it make sense to name it hasDynamicAttributes?

Reply:
I see this from different dimensions:

1. In theory, dynamic attributes are not the only deciding factor for whether this will be GPE'd. For example, Mingsheng showed earlier that some dynamic attributes can also be GPE'd into a graph using attributes on functions.

2. This flag is really about whether we GPE this particular `graph_op` or not. It'd also helps us write some clever tests to assess the benefits of writing certain ops, and implement some heuristics that make parts of the program not be clustered.